batch/batch/resource_usage.py

```diff
@@ -65,19 +77,17 @@ def _decode_to_df(data: bytes) -> Optional[pd.DataFrame]:
 
     def __init__(
         self,
-        container_name: str,
-        container_overlay: str,
-        io_volume_mount: Optional[str],
-        veth_host: str,
+        container: 'Container',
```

Comment:
I detail this below, but this feels like an abstraction violation. The Container and ResourceUsageMonitor classes are now fully entangled.